common/go-utils/src/inv.rs

```diff
@@ -70,19 +62,61 @@ struct GoRelease {
 struct GoFile {
     os: String,
     arch: String,
+    filename: String,
     sha256: String,
     version: String,
 }
 
-impl GoFile {
-    fn target_arch(&self) -> Option<String> {
-        if self.os.is_empty() || self.arch.is_empty() {
-            return None;
-        }
-        Some(format!("{}-{}", self.os, self.arch))
+#[derive(thiserror::Error, Debug)]
+pub enum GoFileConversionError {
+    #[error(transparent)]
+    Version(#[from] VersionParseError),
+    #[error("Arch is not supported: {0}")]
+    Arch(String),
+    #[error("OS is not supported: {0}")]
+    Os(String),
+}
+
+impl TryFrom<&GoFile> for Artifact {
+    type Error = GoFileConversionError;
+
+    fn try_from(value: &GoFile) -> Result<Self, Self::Error> {
+        Ok(Artifact {
+            go_version: value.version.clone(),
+            semantic_version: Version::parse_go(&value.version)?,
+            os: parse_go_os(&value.os)?,
+            arch: parse_go_arch(&value.arch)?,
+            sha_checksum: value.sha256.clone(),
+            url: format!("{}/{}", GO_HOST_URL, value.filename),
+        })
+    }
+}
+
+fn parse_go_arch(arch: &str) -> Result<String, GoFileConversionError> {
+    match arch {
+        "amd64" => Ok(String::from("x86_64")),
+        "arm64" => Ok(String::from("aarch64")),
+        _ => Err(GoFileConversionError::Arch(arch.to_string())),
     }
 }
 
+fn parse_go_os(os: &str) -> Result<String, GoFileConversionError> {
```

Comment:
See my earlier comment because the same applies here. Instead of returning a `Result<String, ...>` you could return a `Result<GoOs, ...>` where `GoOs` is a union type that can only be "linux" (for now).